maven-archetype-plugin/src/main/java/org/apache/maven/archetype/ui/generation/DefaultArchetypeGenerationConfigurator.java

```diff
@@ -313,46 +308,24 @@ else if ( !archetypeGenerationQueryer.confirmConfiguration( archetypeConfigurati
         request.setProperties( properties );
     }
 
-    private String getTransitiveDefaultValue( String defaultValue, ArchetypeConfiguration archetypeConfiguration,
-                                              String requiredProperty, Context context )
+    private static String expandEmbeddedTemplateExpressions( String originalText, String textDescription, Context context )
     {
-        String result = defaultValue;
-        if ( null == result )
+        if ( StringUtils.contains( originalText, "${" ) )
```

Comment:
could be originalText.contains

In java 7+ we no longer need StringUtils.contains

Reply:
`null` safety